client/client_pool/src/external_client.cpp

```diff
@@ -177,25 +177,33 @@ std::pair<int32_t, ConcordClient::PendingReplies> ConcordClient::SendPendingRequ
     request_queue.push_back(bft::client::WriteRequest{single_config, std::move(req.request)});
   }
   try {
-    auto res = new_client_->sendBatch(request_queue, batch_cid);
-    for (const auto& rep : res) {
-      auto cid = seq_num_to_cid.find(rep.first)->second;
-      auto data_size = rep.second.matched_data.size();
-      for (auto& reply : pending_replies_) {
-        if (reply.cid != cid) continue;
-        if (reply.cb) {
-          auto result = bftEngine::SendResult{rep.second};
-          reply.cb(std::move(result));
-          LOG_DEBUG(logger_, "Request processing via callback completed " << KVLOG(client_id_, batch_cid, reply.cid));
+    LOG_INFO(logger_, "Batch processing started" << KVLOG(client_id_, batch_cid));
+    auto receivedRepliesMap = new_client_->sendBatch(request_queue, batch_cid);
+    for (const auto& receivedReplyEntry : receivedRepliesMap) {
+      auto cid = seq_num_to_cid.find(receivedReplyEntry.first)->second;
```

Comment:
May result in a crash if seq_num to cid entry is not found. 
Should we first check if entry is found and then access the value (to be on safe side)?

Reply:
OK